tests/phpunit/tests/query/commentFeed.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/**
+ * @group query
+ */
+class Tests_Query_CommentFeed extends WP_UnitTestCase {
+	public static $post_type = 'post'; // Can be anything.
+
+	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
+		for ( $i = 0; $i < 10; $i ++ ) {
+			$post_id = $factory->post->create(
+				array(
+					'post_type'   => self::$post_type,
+					'post_status' => 'publish',
+				)
+			);
+			$factory->comment->create_post_comments( $post_id, 5 );
+		}
+
+		update_option( 'posts_per_rss', 100 );
+	}
+
+	/**
+	 * @ticket 36904
+	 */
+	public function test_archive_comment_feed() {
+		global $wpdb;
+		add_filter( 'split_the_query', '__return_false' );
+		$q1   = new WP_Query();
+		$args = array(
+			'withcomments'           => 1,
+			'feed'                   => 'comments-rss',
+			'post_type'              => self::$post_type,
+			'update_post_meta_cache' => false,
+			'update_post_term_cache' => false,
+			'ignore_sticky_posts'    => false,
+			'no_found_rows'          => true,
+		);
+		$q1->query( $args );
+		$num_queries = $wpdb->num_queries;
+		$q2          = new WP_Query();
+		$q2->query( $args );
+		$this->assertTrue( $q2->is_comment_feed() );
+		$this->assertFalse( $q2->is_singular() );
```

Comment:
Not sure I understand the need to split these tests up? Can you explain why we would have that overhead?

Reply:
True, there will be a small amount of overhead but the use of shared fixtures in `wpSetUpBeforeClass()` will reduce most of  that.

The reason I suggested splitting them up (or at least doing so differently) is that it will provide more information to developers if something is broken.

In this test, if `$q2->is_comment_feed() === false` then the test to ensure caching is running correctly won't run. If a code change has broken both, it would be helpful to know all at once.

Reply:
This is not about fixtures, it is about what we are testing. We are testing the cache, checking that `is_comment_feed` and `is_singular` are just ensuring that values I submitted to `WP_Query` forced those conditionals to be true. Checking all those values at once, is the value in test. Without the asserts apart, the test is not as useful.

Reply:
OK